src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -3232,14 +3232,10 @@ def _remove_osa_nulls(managed_clusters):
     by the server, but get recreated by the CLI's own "to_dict" serialization.
     """
     attrs = ['tags', 'plan', 'type', 'id']
-    ap_master_attrs = ['name', 'os_type']
     for managed_cluster in managed_clusters:
         for attr in attrs:
             if getattr(managed_cluster, attr, None) is None:
```

Comment:
```suggestion
            if hasattr(managed_cluster, attr) and getattr(managed_cluster, attr) is None:
```

arguably ^ is better, but it doesn't really matter.

Reply:
If remember Python correctly `getattr(managed_cluster, attr)` still needs to be `getattr(managed_cluster, attr, None)`. Otherwise there will be an exception in case when there is no attribute.

Reply:
But it makes sense. I'll update.

Reply:
Ahh. I see... `hasattr` :) Tired brain.